build.gradle.kts

```diff
@@ -174,37 +157,73 @@ subprojects {
 
 	// Source encoding
 	tasks {
+
+		withType<ManifestTask> {
```

Comment:
```suggestion
		withType<ManifestTask>().configureEach {
```
We should use the `.configureEach` here in order to use a lazy configuration.

Separate question:
Now that we are adding an output, this task will have an up-to-date checking mechanism. Before we allow that, are we sure that _all_ inputs and _all_ outputs for this ManifestTask are properly declared?

Reply:
I don't see where any of the inputs such as group, version, scm, date, etc are tracked as inputs here. You mentioned that they were inputs in this comment: https://github.com/twitch4j/twitch4j/pull/715#discussion_r1093433608

Without that, I recommend not declaring the output directory for now.

Reply:
Yep, this data is not tracked as inputs.
https://github.com/coditory/gradle-manifest-plugin/blob/master/src/main/kotlin/com/coditory/gradle/manifest/ManifestAttributes.kt

It's parsed from git commands output and system properties.

Removing the output directory declaration will cause `javadoc` and `jar` tasks to fail because of the detected undeclared dependency of the metadata file.
`manifest` task
<img width="537" alt="Screenshot 2023-02-01 at 9 48 10 AM" src="https://user-images.githubusercontent.com/1287797/216123029-1ef12f8c-c34e-4e35-859c-7dddd9f429c1.png">
`jar task`
<img width="643" alt="Screenshot 2023-02-01 at 9 49 04 AM" src="https://user-images.githubusercontent.com/1287797/216123034-b0b98511-bf94-4001-a79a-11fd48bed6ba.png">

Reply:
This said, one of the options I see is to disable the manifest plugin for all use cases except for publishing. 
This is an idea pretty similar to what Android does by default - having separate `debug` and `release` configs for the build.
e.g. 
```kotlin
if (project.properties.containsKey("enableManifest")) 
  apply(plugin = "com.coditory.manifest")
```
```shell
./gradlew publish -PenableManifest
```

@iProdigy what do you think?

Reply:
@stepango thanks for the pull request and the investigation :)

Disabling the plugin outside of `publish` seems like a good idea. (until the plugin is fixed)
Would the following change work to automate the detection without a new property?

`project.properties.containsKey("enableManifest")`
->
`project.gradle.startParameter.taskNames.any { s -> s.startsWith("publish") }`

Reply:
This will work as long as you are not using short names for publish task e.g. `pub` :)

Reply:
I've ended up adding both options
``` kotlin
project.properties.containsKey("enableManifest") ||
  project.gradle.startParameter.taskNames.any { s -> s.startsWith("publish") }
```

I've kept the flag for testing purposes, e.g. you want to see what is inside the manifest files without actually publishing 

@iProdigy @PhilippHeuer let me know if this works for you.